src/tests/GC/Stress/Framework/ReliabilityFramework.cs

```diff
@@ -270,6 +283,8 @@ public static int Main(string[] args)
 
         GC.Collect(2);
         GC.WaitForPendingFinalizers();
+        
+        Environment.Exit(0);
```

Comment:
Is this intentional?

Reply:
Yes, it is. The intention is to make sure when the main thread exits, the process is shut down. Without that, the process would wait until all the started test cases are completed. That's okay, but it makes scripting difficult. I would like to be able to cascade runs with different GC configs and have a predictable finishing time.

If we want to wait until the test cases are completed, we can use the `debugBreakOnTestHang` flag (which is (and was) on by default). Without a debugger attached, that flag will make sure the main thread waits.

Reply:
Maybe we could mark the test threads as background instead. That would have the same effect.